### DISCUSSION TITLE
Trouble with input gpio_keys with device runtime power management

### DISCUSSION THREAD

**Discussion by hglassdyb** [zephyrproject-rtos/zephyr#69900] [Q&A]
Hello,

I’m trying to migrate a project from Zephyr 3.5 to Zephyr 3.6 and I encountered some issues with the input gpio_keys driver that is used for some buttons:

1. The keys are not responding to presses when using `CONFIG_PM_DEVICE=y`.

    - The problem is that the buttons are now suspended by default by the device runtime power management, and therefore they are not sending the input events. 

2. One of the keys was stuck in the high state (1).

    - The gpio keys interrupt were sending events during the initialization, and one of them stayed high (1), as if the button was always pressed. This happens even if none of the keys are pressed.
    - It seems to come from the fact that the pin is disabled before the interruption is disabled in the `gpio_keys_pm_action()`.
    - The pin is the only one with `GPIO_ACTIVE_LOW` flag, which could also be related to the issue.

So I managed to solve both of these issues with some modifications on the driver, but I'm not sure if this is a bug or a misuse of Zephyr on my side.

Here are the modifications:

```diff
diff --git a/drivers/input/input_gpio_keys.c b/drivers/input/input_gpio_keys.c
index cd277c19fdc..ee50156893b 100644
--- a/drivers/input/input_gpio_keys.c
+++ b/drivers/input/input_gpio_keys.c
@@ -197,6 +197,12 @@ static int gpio_keys_init(const struct device *dev)
 		return ret;
 	}
 
+	ret = pm_device_runtime_get(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to resume device");
+		return ret;
+	}
+
 	return 0;
 }
 
@@ -229,6 +235,8 @@ static int gpio_keys_pm_action(const struct device *dev,
 	for (int i = 0; i < cfg->num_keys; i++) {
 		const struct gpio_dt_spec *gpio = &cfg->pin_cfg[i].spec;
 
+		if( gpio_flags == GPIO_INPUT )
+		{
 			ret = gpio_pin_configure_dt(gpio, gpio_flags);
 			if (ret != 0) {
 				LOG_ERR("Pin %d configuration failed: %d", i, ret);
@@ -244,6 +252,23 @@ static int gpio_keys_pm_action(const struct device *dev,
 				LOG_ERR("interrupt configuration failed: %d", ret);
 				return ret;
 			}
+
+		}
+		else {
+			if (!cfg->polling_mode) {
+				ret = gpio_pin_interrupt_configure_dt(gpio, int_flags);
+				if (ret < 0) {
+					LOG_ERR("interrupt configuration failed: %d", ret);
+					return ret;
+				}
+			}
+
+			ret = gpio_pin_configure_dt(gpio, gpio_flags);
+			if (ret != 0) {
+				LOG_ERR("Pin %d configuration failed: %d", i, ret);
+				return ret;
+			}
+		}
 	}
 
 	if (action == PM_DEVICE_ACTION_RESUME && cfg->polling_mode) {
```

**Comment by fabiobaltieri** [comment]
Hey thanks for reporting, looks like an actual issue, I've converted into an issue in https://github.com/zephyrproject-rtos/zephyr/issues/70068, let's followup there.
